ms-common/src/main/java/net/trajano/ms/core/ConstraintViolationResponse.java

```diff
@@ -0,0 +1,130 @@
+import javax.xml.bind.annotation.XmlRootElement;
+import javax.xml.bind.annotation.XmlType;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonInclude.Include;
+
+import io.swagger.annotations.ApiModelProperty;
+
+/**
+ * Contraint violation response. This is the error response that is sent when
+ * there is a Bean Validation error on the input data.
+ *
+ * @author Archimedes Trajano
+ */
+@XmlRootElement
+@XmlType(propOrder = {
+    "error",
+    "errorDescription",
+    "errorClass",
+    "violations",
+    "requestId",
+    "requestUri",
+    "threadId",
+    "host",
+    "jwtId",
+    "stackTrace",
+    "cause"
+})
+@JsonInclude(Include.NON_NULL)
+public class ConstraintViolationResponse extends ErrorResponse {
+
+    @XmlAccessorType(XmlAccessType.PROPERTY)
+    @XmlType(propOrder = {
+        "message",
+        "messageTemplate",
+        "path"
+    })
+    @SuppressWarnings("unused")
+    private static class ConstraintViolationElement {
+
+        /**
+         * Prefix added to the path.
+         */
+        private final String pathPrefix;
+
+        /**
+         * The wrapped violation.
+         */
+        private final ConstraintViolation<?> violation;
+
+        ConstraintViolationElement(final ConstraintViolation<?> violation) {
+
+            this.violation = violation;
+            pathPrefix = "";
+        }
+
+        ConstraintViolationElement(final ConstraintViolation<?> violation,
+            final String pathPrefix) {
+
+            this.violation = violation;
+            this.pathPrefix = pathPrefix + ".";
+        }
+
+        public String getMessage() {
+
+            return violation.getMessage();
+        }
+
+        public String getPath() {
+
+            return pathPrefix + violation.getPropertyPath().toString();
+        }
+
+        public String getTemplate() {
+
+            return violation.getMessageTemplate();
+        }
+    }
+
+    /**
+     * The request ID. This is obtained from the header.
+     */
+    @ApiModelProperty(name = "violations",
+        value = "Constraint violations.")
+    @XmlElement(name = "violations")
+    private final List<ConstraintViolationElement> violations;
+
+    /**
+     * Wraps a {@link Throwable} in an {@link ErrorResponse} with full stack trace
+     * and cause if requested.
+     *
+     * @param e
+     *            constraint violation exception to wrap
+     * @param uriInfo
+     *            URI info
+     * @param showStackTrace
+     *            flag to determine whether the stack trace is to be shown.
+     */
+    public ConstraintViolationResponse(final ConstraintViolationException e,
+        final UriInfo uriInfo,
+        final boolean showStackTrace) {
+
+        super(e, uriInfo, showStackTrace);
+        violations = StreamSupport.stream(e.getConstraintViolations().spliterator(), false).map(v -> new ConstraintViolationElement(v)).collect(Collectors.toList());
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this lambda with a method reference. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1612)